Added new community plugin - Quick Latex for Obsidian

joeyuping opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/joeyuping/quick_latex_obsidian

## Release Checklist

<!--- Confirm that you have done the following before submitting your plugin -->
- I have tested this on:
  - [X]  Windows
  - [x] macOS
  - [x] Linux
- [X] My GitHub release contains all required files
  - [X] `main.js`
  - [X] `manifest.json`
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] README clearly describes the plugins purpose and provides clear usage instructions.
- [X] I have added a license in the LICENSE file.

lishid commented:
https://github.com/joeyuping/quick_latex_obsidian/blob/d4bf584c781e0b7413d376622df02092567fa8b3/main.ts#L14
Don't keep this list as it will cause a memory leak. Instead, you can use `workspace.iterateCodeMirrors` to go through all currently active CodeMirror instances in your plugin onunload.

https://github.com/joeyuping/quick_latex_obsidian/blob/d4bf584c781e0b7413d376622df02092567fa8b3/main.ts#L37
I feel like these should be editor commands rather than a direct CodeMirror hook.

https://github.com/joeyuping/quick_latex_obsidian/blob/d4bf584c781e0b7413d376622df02092567fa8b3/main.ts#L88
https://github.com/joeyuping/quick_latex_obsidian/blob/d4bf584c781e0b7413d376622df02092567fa8b3/main.ts#L128
activeLeaf may be null causing a crash here - use `workspace.getActiveViewOfType(MarkdownView)` and check for null before continuing.

joeyuping commented:
@lishid Thank you so much for the kind feedbacks!

I have incorporated your suggestions, and also added a settings page.  New files have been published in the same repository as 1.3.0.
